parcels/fieldfilebuffer.py

```diff
@@ -35,6 +35,7 @@ def __init__(
         self.cast_data_dtype = kwargs.pop("cast_data_dtype", np.float32)
         self.ti = None
         self.interp_method = interp_method
+        self.gridindexingtype = kwargs.pop("gridindexingtype", "nemo")
```

Comment:
Should we update `_FileBuffer` signature and associated init?

```diff

class _FileBuffer:
    def __init__(
        self,
        filename,
        dimensions,
        indices,
        timestamp=None,
        interp_method: InterpMethodOption = "linear",
        data_full_zdim=None,
-        **kwargs,
+       cast_data_dtype = np.float32,
+       gridindexingtype = "nemo",
    ):
    ...
```
I would like to shift away from using `kwargs` where named arguments are better suited (i.e, unless we're using them to propagate through to other functions, or need arbitrarily named keyword arguments. [SO post talking about it](https://stackoverflow.com/a/1415830/15545258)

Reply:
OK, but should we do that here, or in another PR/refactor? Note that the `DaskFileBuffer(NetcdfFileBuffer)` also has some `kwargs.pop()` calls; and purest breaks if we don't put these also in the `_FileBuffer()` class.
https://github.com/OceanParcels/Parcels/blob/c376f0d2056f1bafbd65b08acef6266732a69dd3/parcels/fieldfilebuffer.py#L304-L307

So it's a bit more work than just this change?

Reply:
yeah, I think migrating away from `kwargs` entirely for all classes would be difficult given some of the inheritence structures that we have. This comment was mainly for the `_FileBuffer()` which I think is a simple fix.

Other changes here can be left for future refactorings (which will hopefully also include removing this large inheritence structure, as it results in a lot of coupling between the classes -- I think a composition approach would be worth exploring)